props/saber_fett263_buttons.h

```diff
@@ -293,248 +662,3382 @@ class SaberFett263Buttons : public virtual PropBase {
+        case 3:
+          sound_queue_.Play(SoundToPlay("thirty.wav"));
+          break;
+        case 4:
+          sound_queue_.Play(SoundToPlay("forty.wav"));
+          break;
+        case 5:
+          sound_queue_.Play(SoundToPlay("fifty.wav"));
+          break;              
+        case 6:
+          sound_queue_.Play(SoundToPlay("sixty.wav"));
+          break;  
+        case 7:
+          sound_queue_.Play(SoundToPlay("seventy.wav"));
+          break;
+        case 8:
+          sound_queue_.Play(SoundToPlay("eighty.wav"));
+          break;  
+        case 9:
+          sound_queue_.Play(SoundToPlay("ninety.wav"));
+          break;  
+      }
+      if (number != 0) sound_queue_.Play(SoundToPlay(&SFX_mnum, number - 1));
+    }
+  }
+
+  void SayNumber (float number, SayType say_type) {
+    int thousand = ((int)floorf(number / 1000)) % 100;
+    int hundred = ((int)floorf(number / 100)) % 10;
+    int ones = ((int)floorf(number)) % 100;
+    int tenths = ((int)floorf(number * 10)) % 10;
+    int hundredths = ((int)floorf(number * 100)) % 10;
+    switch (say_type) {
+      case SAY_BATTERY:
+      case SAY_DECIMAL:
+        // Battery Level prompt
+        if (say_type == SAY_BATTERY) sound_queue_.Play(SoundToPlay("mbatt.wav"));
+        // Tens & Ones
+        if (number == 0) {
+          sound_queue_.Play(SoundToPlay("mzero.wav"));
+        } else {
+          TensValue(ones);
+        }
+        // Decimal / Point
+        sound_queue_.Play(SoundToPlay("mpoint.wav"));
+        // Tenths
+        if (tenths == 0) {
+          sound_queue_.Play(SoundToPlay("mzero.wav"));
+        } else {
+          sound_queue_.Play(SoundToPlay(&SFX_mnum, tenths - 1));
+        }
+        // Hundredths
+        if (hundredths != 0) sound_queue_.Play(SoundToPlay(&SFX_mnum, hundredths - 1));
+        // Volts prompt
+        if (say_type == SAY_BATTERY) sound_queue_.Play(SoundToPlay("mvolts.wav"));
+        break;
+      case SAY_WHOLE:
+      case SAY_MILLIS:
+        // Thousands
+        if (thousand > 0) {
+          TensValue(thousand);
+          sound_queue_.Play(SoundToPlay("thousand.wav"));
+        }
+        // Hundred
+        if (hundred > 0) {
+          sound_queue_.Play(SoundToPlay(&SFX_mnum, hundred - 1));
+          sound_queue_.Play(SoundToPlay("hundred.wav"));          
+        }
+        // Tens & Ones
+        if (ones == 0) {
+          if ((thousand + hundred) == 0) sound_queue_.Play(SoundToPlay("mzero.wav"));
+        } else {
+          TensValue(ones);
+        }
+        if (say_type == SAY_MILLIS) sound_queue_.Play(SoundToPlay("mmillis.wav"));
+        break;
+      default:
+        break;
+    }
+  }
+   
+  enum ClashType {
+    CLASH_NONE,
+    CLASH_CHECK,
+    CLASH_BATTLE_MODE,
+    CLASH_LOCKUP,
+    CLASH_LOCKUP_END,
+#ifdef FETT263_CLASH_STRENGTH_SOUND    
+    CLASH_NORMAL,
+    CLASH_STAB,
+#endif
+  };
+
+#ifdef FETT263_CLASH_STRENGTH_SOUND
+  void SelectFloat(Effect* effect, float value) {
```

Comment:
This function should be in the Effect class.

Reply:
I've been doing all of the edits on github.com and there doesn't appear to be a search-and-replace.  I was concerned that if I just copy all of the code from the form on github.com and use Find and Replace in an external text editor then it would no longer keep track of what was changed. 

So I should move SelectFloat() to sound/effect.h?

Reply:
moved SelectFloat() to sound/effect.h